src/app/COMPONENTS/task-details/task-details.component.ts

```diff
@@ -31,11 +34,60 @@ export class TaskDetailsComponent implements OnInit {
     createdOn: new FormControl({ value: "", disabled: true }, Validators.required),
   })
 
-
   constructor(private tasksService: TasksService, private activatedRoute: ActivatedRoute, private fb: FormBuilder) { }
 
   ngOnInit() {
-      this.task$ =  this.activatedRoute.params.pipe(map((params) => params['id'] as string),
-      switchMap((taskId) => this.tasksService.getTaskById(taskId)))
+    this.activatedRoute.params.pipe(map((params) => params['id'] as string),
+      switchMap((taskId) => this.tasksService.getTaskById(taskId)), take(1)
+    ).subscribe(task => {
+      this.tasksService.taskSubject.next(task)
+      this.taskId = task.id
+      this.taskForm.patchValue({
+        title: task.title,
+        description: task.description,
+        type: task.type,
+        status: task.status,
+        createdOn: task.createdOn
+      })
+    })
+  }
+
+  updateTaskDetails() {
+    const updatedTaskDetails = this.taskForm.value
+
+    this.tasksService.updateTask(updatedTaskDetails, this.taskId).pipe(
+      take(1)
+    ).subscribe({
+      next: task => {
+        this.tasksService.taskSubject.next(task)
+        this.editing = false
+        this.taskForm.get('title')?.disable()
+        this.taskForm.get('description')?.disable()
+        this.taskForm.get('status')?.disable()
+        this.taskForm.get('type')?.disable()
+        this.taskForm.patchValue({
+          title: task.title,
+          description: task.description,
+          type: task.type,
+          status: task.status,
+          createdOn: task.createdOn
+        })
+      },
+      error: err => {
+        this.editing = true
+        this.taskForm.get('title')?.enable()
```

Comment:
change this to use the allowTaskEditing method, to reduce redundacy of the same code